src/e2e-test/java/io/cdap/plugin/salesforcebatchsource/stepsdesign/DesignTimeSteps.java

```diff
@@ -76,4 +88,35 @@ public void clickOnNewServiceNowConnection() {
     SalesforcePropertiesPageActions.clickOnServicenowConnection();
   }
 
+  @Then("Validate the values of records transferred to target Big Query table is equal to the values from source table")
+  public void validateTheValuesOfRecordsTransferredToTargetBigQueryTableIsEqualToTheValuesFromSourceTable()
+    throws IOException, InterruptedException, IOException, SQLException, ClassNotFoundException, ParseException {
+    int targetBQRecordsCount = BigQueryClient.countBqQuery(PluginPropertyUtils.pluginProp("bqTargetTable"));
+    BeforeActions.scenario.write("No of Records Transferred to BigQuery:" + targetBQRecordsCount);
+    Assert.assertEquals("Out records should match with target BigQuery table records count",
+                        CdfPipelineRunAction.getCountDisplayedOnSourcePluginAsRecordsOut(), targetBQRecordsCount);
+
+    boolean recordsMatched = BQValidation.validateSalesforceToBQRecordValues
+      (PluginPropertyUtils.pluginProp("customObject"),
+       PluginPropertyUtils.pluginProp("bqTargetTable"));
+    Assert.assertTrue("Value of records transferred to the target table should be equal to the value " +
+                        "of the records in the source table", recordsMatched);
+  }
+  @Then("Read Credentials file {string} with value: {string}")
```

Comment:
You have to remove this method when you are raising the PR, same goes for the step. You can create this method in a different class for local testing but don't push that class.

Reply:
done